### ISSUE TITLE
Compile and distribute binary in release pages

### ISSUE THREAD

**Issue by networm** [Koihik/LuaFormatter#28]
I have test this tool on macOS, it has a good format quality. I think it should be precompiled on major operating systems: Windows, macOS, Linux.
I think it should help people using this with no prerequisites such as Visual Studio, Xcode, clang and so on.

So would you add a continuous integration tool such as [Travis CI](https://travis-ci.org/) or [AppVeyor](https://www.appveyor.com/) to build and release?

**Comment by Koihik** [comment]
We do not provide binary in release page after support common editors(VSC/Sublime/vim). But you can still find those binary in [editor plugin](https://github.com/Koihik/vscode-lua-format/tree/master/bin).

**Comment by networm** [comment]
Thank you for mentioning that I can find binary in editor plugin page.
But I still think the binary files should include in release. Because if trying to integrate this tool to continuous integration or git pre-commit hook, we still need binary to execute.

**Comment by AndreMiras** [comment]
It's on old topic, but still valid.
I started using LuaFormatter recently and it's a pretty good tool. However it take quite a while to build in our CI.
That would save a lot of accumulated time across developers using it if we were publishing pre-built binary to LuaRocks for different platforms and that would help it to become a defacto standard making it also more suitable for people to build reusable GitHub Actions for it.
I'm new to Lua so I don't know how achievable it is, but I've done it a few times for Python, last time was for [lru-dict](https://github.com/amitdev/lru-dict/commits?author=AndreMiras).
Basically the idea is to (cross-)compile the library for the different platform using the CI.
But we could start with only building it for the simple platforms and then go more advanced with cross compiling.
I'm also pretty familiar with cross compiling and CI as I helped setting up automated testing on python-for-android and kivy-ios in the past.
I would be glad to help if we're willing to move forward with this and if I get some guidance.
